docs/sources/operations/caching.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Caching 
+menuTitle: Caching 
+description: Enable and configure memcached for caching. 
+weight: 100
+keywords:
+  - memcached
+  - caching
+---
+<!-- For more information about how to populate front matter, see [Topic front matter]({{< relref "../../front-matter/" >}}). -->
+
+# Caching
+
+Loki supports caching of index writes and lookups, chunks and query results to
+speed up query performance. This sections describes the recommended Memcached
+configuration to enable caching for chunks and query results. The index lookup
+cache is configured to be in-memory by default.
+
+## Before you begin
+
+- It is recommended to deploy four dedicated Memcached clusters.
+- As of 2023-02-01 the `memcached:1.6.17-alpine` is recommended.
+- Consult the Loki ksonnet [memcached](https://github.com/grafana/loki/blob/main/production/ksonnet/loki/memcached.libsonnet) deployment and the ksonnet [memcached library](https://github.com/grafana/jsonnet-libs/tree/master/memcached).
+
+## Steps
+
+To enable and configure Memcached:
+
+1. Deploy each Memcached service with at least three replicas and configure
+   each as follows:
+    1. Chunk cache `--memory-limit=4096 --max-item-size=2m --conn-limit=1024`
+    1. Query result cache `--memory-limit=1024 --max-item-size=5m --conn-limit=1024`
+
+1. Configure Loki to use the cache.
+    1. If the Helm chart is used
+        1. set `memcached.chunk_cache.host` to the Memecache address of for the chunk cache.
+        1. set `memcached.results_cache.host` to the Memecache address of for the query result cache.
+	1. set `memcached.chunk_cache.enabled=true` and `memcached.results_cache.enabled=true`.
+        1. Ensure that the connection limit of Memcached is at least `number_of_clients? * max_idle_conns`.
+  1. If the Loki configuration is used
+      1. Configure the chunk cache
+      ```yaml
+      chunk_store_config:
+        chunk_cache_config:
+          memcached:
+            batch_size: 256
+            parallelism: 10
+          memcached_client:
+            host: <memcached host>
+            addresses: ???
+            service: ???
```

Comment:
@trevorwhitney what should addresses and service be?

Reply:
So both, `host` and `service` must be specified?

Reply:
I believe so. We use `memcached-client` in cloud so I'd stick with that as the default.

Reply:
> From the docs addresses is experimental, I would suggest using the host and service fields.
> 
> host: I have pointed this to the k8s service that is deployed for memcached service: Is the port name on that service exposing memcached
> 
> I have deployed 4 x memcached
> 
> ```
> helm upgrade --install chunk-cache -n loki bitnami/memcached -f memcached-overrides.yaml
> helm upgrade --install write-cache -n loki bitnami/memcached -f memcached-overrides.yaml
> helm upgrade --install results-cache -n loki bitnami/memcached -f memcached-overrides.yaml
> helm upgrade --install index-cache -n loki bitnami/memcached -f memcached-overrides.yaml
> ```
> 
> ```
> loki:
>   memcached:
>     chunk_cache:
>       enabled: true
>       host: chunk-cache-memcached.loki.svc
>       service: memcache
>       batch_size: 256
>       parallelism: 10
>     results_cache:
>       enabled: true
>       host: results-cache-memcached.loki.svc
>       service: memcache
>       default_validity: 12h
> 
>   structuredConfig:
>     chunk_store_config:
>       write_dedupe_cache_config:
>         memcached:
>           batch_size: 100
>           parallelism: 100
>         memcached_client:
>           host: write-cache-memcached.loki.svc
>           service: memcache
>           consistent_hash: true
> 
>     storage_config:
>       index_queries_cache_config:
>         memcached:
>           batch_size: 100
>           parallelism: 100
>         memcached_client:
>           host: index-cache-memcached.loki.svc
>           service: memcache
>           consistent_hash: true
> ```

Any chance you can  document this? `memcached-overrides.yaml` 

The bitnami chart being referenced can run standalone, or run in HA mode.  But HA mode would appear to require being able to pass in multiple hosts values to Loki.  So I assume you really want it to run in standalone mode, but then we only have 1 replica, not 3.

Any additional insight would be great.